src/components/common-components/autocomplete/autocomplete.component.tsx

```diff
@@ -52,18 +52,40 @@ export function AutocompleteComponent<T>({
     getOptionLabel,
     open: true,
   });
+  const [IsComponentVisible, setIsComponentVisible] = useState(true);
+  const pageOffset: number = document.getElementById("root")?.children[0].children[0].scrollTop!;
+  let clearModal;
   return (
-    <div ref={inputRef} data-cy="shiftDropdown">
+    <div
+      ref={inputRef}
+      data-cy="shiftDropdown"
+      style={{ display: IsComponentVisible ? "initial" : "none" }}
+      onWheel={(e: React.WheelEvent<HTMLTableCellElement>): void => {
+        pageOffset !== document.getElementById("root")?.children[0].children[0].scrollTop &&
+          setIsComponentVisible(false);
+      }}
+      onMouseEnter={(): void => {
+        clearTimeout(clearModal);
```

Comment:
Spoko pomysł z tym timerem, ale trochę się obawiam wycieku pamięci. Zawsze kiedy spotykałem się z timerem to był on w useEffecie, który miał returna czyszczącego ten timer w przypadku odmontowania komponenetu. 

Obecnie może nastąpić taka sytuacja, że timer nie zostanie wyczyszczony, jeśli komponent zostanie odmontowany przed odpaleniem eventu onMouseEnter.